### ISSUE TITLE
Support WC 3.0

### ISSUE THREAD

**Issue by franticpsyx** [Prospress/woocommerce-subscribe-all-the-things#96]
Opening this to start working on WC 2.7 compatibility.

Since SATT does not extend any core CRUD classes, WC 2.7 compatibility is limited to the following tasks (more added as we investigate):

- Stop accessing CRUD object properties directly.
- Stop accessing deprecated object properties.
- Look for deprecated CRUD object functions and use new ones where applicable.
- Replace deprecated filters with new equivalents.
- Phase out use of post/post meta functions.
- Use CRUD API to save plugin data on product objects.

**Comment by franticpsyx** [comment]
Switching milestone to 1.2. 

Reason: Scheme data will be obtained via the Product API in 1.2, which will abstract a lot of direct `get_post_meta` calls. Once this is done, scheme data will be obtained using `WC_Data::get_meta`.

**Comment by franticpsyx** [comment]
See https://github.com/Prospress/woocommerce-subscribe-all-the-things/tree/issue-96

Leaving this open until we can test with a final WC 2.7-compatible version of WCS.

**Comment by franticpsyx** [comment]
On with this :)

**Comment by franticpsyx** [comment]
Quick update:

If you depend on SATT, it is recommended to stick with WC 2.6 until a final SATT 2.0 release is available. Feel free to take the `2.0` branch for a spin and let me know what works / doesn't work under WC 3.0 / WCS 2.2 . Thanks!

See progress in https://github.com/Prospress/woocommerce-subscribe-all-the-things/compare/master...2.0

**Comment by franticpsyx** [comment]
Note: Integrations with PB, CP and MnM do not work in `2.0` just yet, as they need to be revised to utilize the new [Products API](https://github.com/Prospress/woocommerce-subscribe-all-the-things/blob/2.0/includes/class-wcsatt-product.php) introduced in SATT 2.0

**Comment by BobBrokman** [comment]
Hi, 
I was trying to evaluate this plugin which looked pretty cool.
I checked out branch 2.0 and gave it a run alongside WC 3
As you can see from the screenshot (https://ibb.co/hsK8Cv), if I select 24 months subscription for the whole cart, it still shows the full price(s). Is it a bug or there are some extra settings I need to tweak?

**Comment by sambomcd** [comment]
Hi,

I know you guys are super busy with the 2.0 update (huge thanks!), and I don't want to waste your time. I've been trying 2.0 out. Forgive the vagueness, but pre2.0 if I was subscribed to a Product Bundle, and the items bundled within that changed between renewals, then the next renewal would give me the updated contents of the Bundle. I am seeing in 2.0 that if I change the items in the Bundle, each renewal just keeps giving me the same contents as when i first signed up, even if those products have since been removed from the Bundle.

May be a unique user case, and I know that PB's are not quite supported in 2.0 yet, but I thought it worth asking, otherwise I am headed in the wrong direction bigly.

**Comment by franticpsyx** [comment]
@sambomcd Can you please open a new issue about this? Thanks!

**Comment by sambomcd** [comment]
Of course! Apologies - I just wasn't sure if this was an actual issue or if I am confused about how it is supposed to work. Normally it is the latter...
